js/fileinput.js

```diff
@@ -124,6 +130,93 @@
     }
   },
 
+  Fileinput.prototype.setImageTransform = function($img, file) {
+      var Fileinput = this;
+      var reader = new FileReader();
+      reader.onload = function(me) {
+        var transform = false;
+        var view = new DataView(reader.result);
+        var exif = Fileinput.getImageExif(view);
+        if (exif) {
+            Fileinput.resetOrientation($img, exif);
+        }
+      }
+
+      reader.readAsArrayBuffer(file);
+  }
+
+  Fileinput.prototype.getImageExif = function(view) {
+    if (view.getUint16(0, false) != 0xFFD8) { 
+      return -2;
+    }    
+    var length = view.byteLength, offset = 2;
+    while (offset < length) {
+      var marker = view.getUint16(offset, false);
+          offset += 2;
+      if (marker == 0xFFE1) {
+        if (view.getUint32(offset += 2, false) != 0x45786966) { 
+          return -1;
+        }
+        var little = view.getUint16(offset += 6, false) == 0x4949;
+            offset += view.getUint32(offset + 4, little);
+        var tags = view.getUint16(offset, little);
+            offset += 2;
+        for (var i = 0; i < tags; i++)   {
+          if (view.getUint16(offset + (i * 12), little) == 0x0112) {
+            return view.getUint16(offset + (i * 12) + 8, little); 
+          }
+        }
+      }
+      else if ((marker & 0xFF00) != 0xFF00){
+         break;
+      } else {
+        offset += view.getUint16(offset, false);
+      } 
+    }
+
+    return -1;
+  }
+
+  Fileinput.prototype.resetOrientation = function($img, transform) {
+  var img = new Image();    
+  
+  img.onload = function() {
+    var width = img.width,
+        height = img.height,
+        canvas = document.createElement('canvas'),
+        ctx = canvas.getContext("2d");
```

Comment:
Here will be problems with image size. 
I try do this by css first, just add style to the image. In this case images go out from image block, and some part of image are crop by `overflow: hidden`. 
This can be done also by css, but then need set height of image manually from it width when image are rotated by 90/-90 degree.